ui/src/components/button/Button.tsx

```diff
@@ -50,42 +51,42 @@ const textColor = ({ disabled, flat, outlined }: ButtonProps): string => {
   return "text-primary";
 };
 
-const hoverTextColor = ({ outlined }: ButtonProps): string =>
-  outlined ? "hover:text-white" : "";
+const hoverTextColor = ({ outlined }: ButtonProps): string | null =>
+  outlined ? "hover:text-white" : null;
 
-const bgColor = ({ disabled, flat, outlined }: ButtonProps): string => {
-  if (flat || outlined) return "";
+const bgColor = ({ disabled, flat, outlined }: ButtonProps): string | null => {
+  if (flat || outlined) return null;
   if (disabled) return "bg-gray-400";
   return "bg-primary";
 };
 
-const hoverBgColor = ({ disabled, flat }: ButtonProps): string => {
-  if (disabled) return "";
+const hoverBgColor = ({ disabled, flat }: ButtonProps): string | null => {
+  if (disabled) return null;
   if (flat) return "hover:bg-gray-100";
   return "hover:bg-primary-dark";
 };
 
-const border = ({ outlined }: ButtonProps): string =>
-  outlined ? "border" : "";
+const border = ({ outlined }: ButtonProps): string | null =>
+  outlined ? "border" : null;
 
-const borderColor = ({ outlined, disabled }: ButtonProps): string => {
-  if (!outlined) return "";
+const borderColor = ({ outlined, disabled }: ButtonProps): string | null => {
+  if (!outlined) return null;
   if (!disabled) return "border-primary hover:border-primary-dark";
   return "border-gray-400";
 };
 
-const cursorAndPointerEvents = ({ disabled }: ButtonProps): string =>
-  !disabled ? "" : "cursor-default pointer-events-none";
+const cursorAndPointerEvents = ({ disabled }: ButtonProps): string | null =>
+  disabled ? "cursor-default pointer-events-none" : null;
 
-const padding = ({ icon }: ButtonProps): string => (icon ? "" : "px-3.5 py-2");
+const padding = ({ icon }: ButtonProps): string | null => (icon ? null : "px-3.5 py-2");
```

Comment:
Maybe insert all the short if-s directly to the `classNames`, as `{ "class-name": condition }`